### ISSUE TITLE
Transform falls back to direct1 Method when toroidal resolution N>=20

### ISSUE THREAD

**Issue by dpanici** [PlasmaControl/DESC#358]
need to still find exactly where/why this is happening, but when using `Equilibrium.from_near_axis`, I find that at high enough toroidal resolution, the transforms for the surface revert to `direct1` method instead of `direct2`:

```python
from qsc import Qsc
from desc.equilibrium import Equilibrium

name =  'precise QA'
qsc = Qsc.from_paper(name)
r=0.1

# works
desc_eq_1 = Equilibrium.from_near_axis(qsc,
                                         r= r,
                                         L=6,
                                         M=6,
                                         N=19,
                                         ntheta=45)
# increment N by 1, throws warning about reverting to direct1
desc_eq_1 = Equilibrium.from_near_axis(qsc,
                                         r= r,
                                         L=6,
                                         M=6,
                                         N=20,
                                         ntheta=45)
```

warning being: 
```python
/home/dpanici/DESC/desc/transform.py:319: UserWarning: direct2 method requires that node pattern is the same on each zeta plane, falling back to direct1 method```

**Comment by dpanici** [comment]
full traceback of warning:

```python
UserWarning                               Traceback (most recent call last)
/tmp/ipykernel_1166/1307181397.py in <module>
      3 warnings.filterwarnings('error')
      4 r=0.1
----> 5 desc_eq = Equilibrium.from_near_axis(qsc,
      6                                          r= r,
      7                                          L=6,

~/DESC/desc/equilibrium/equilibrium.py in from_near_axis(cls, na_eq, r, L, M, N, ntheta, spectral_indexing)
    378 
    379         eq = Equilibrium(**inputs)
--> 380         eq.surface = eq.get_surface_at(rho=1)
    381 
    382         return eq

~/DESC/desc/equilibrium/configuration.py in get_surface_at(self, rho, theta, zeta)
    572             surface.R_lmn = Rb
    573             surface.Z_lmn = Zb
--> 574             surface.grid = LinearGrid(
    575                 rho=rho, M=2 * surface.M, N=2 * surface.N, endpoint=True, NFP=self.NFP
    576             )

~/DESC/desc/geometry/surface.py in grid(self, new)
    161                 f"grid should be a Grid or subclass, or ndarray, got {type(new)}"
    162             )
--> 163         self._R_transform.grid = self.grid
    164         self._Z_transform.grid = self.grid
    165 

~/DESC/desc/transform.py in grid(self, grid)
    671                 self._check_inputs_fft(self.grid, self.basis)
    672             if self.method == "direct2":
--> 673                 self._check_inputs_direct2(self.grid, self.basis)
    674             if self.built:
    675                 self._built = False

~/DESC/desc/transform.py in _check_inputs_direct2(self, grid, basis)
    317             grid.nodes[:, :2].T.reshape((2, zeta_cts[0], -1), order="F")
    318         ):
--> 319             warnings.warn(
    320                 colored(
    321                     "direct2 method requires that node pattern is the same on each "

UserWarning: direct2 method requires that node pattern is the same on each zeta plane, falling back to direct1 method```

**Comment by dpanici** [comment]
Inside of the [`isalmostequal`](https://github.com/PlasmaControl/DESC/blob/362f6b5be5321b7d4ca41035d3f59dd03449cdee/desc/utils.py#L297) function, the multiplication by the number of elements in the array that is being taken the std of (`*x.shape[axis]` in the line`np.all(x.std(axis=axis) * x.shape[axis] < tol)`) is what takes an array that otherwise has a std of like 1e-14 and pushes it past the 1e-12 tol.

Is this multiplication necessary?

**Comment by dpanici** [comment]
Issue is actually due to numpy.std having numerical precision issues, a solution could be do cast to float128 when doing the np.std call, like 
```
np.all(x.std(axis=axis,dtype=np.float128) * x.shape[axis] < tol)
```

**Comment by dpanici** [comment]
Fixed by #359
